### PR TITLE
Adds @solana/wallet-adapter

### PR DESCRIPTION
**Pull request by kierangillen**
## Switch wallet sign on to use @solana/wallet-adapter

- Adds [@solana/wallet-adapter](https://github.com/solana-labs/wallet-adapter) and removes our custom `useWallet` hook. The new wallet adapter package includes a new `useWallet` hook, but we are still relying heavily on the `mangoStore` wallet throughout the codebase. This PR stitches the Solana Labs adapter with our store, which allows us to still use the mango store wallet without introducing a large code update. We should incrementally refactor components to use the new system.

- Note: We are using a default list of wallets `['Phantom', 'Solflare', 'Sollet']` and then if the user has any detected wallet, those will be added to the dropdown. The default list allows for new users to see a recommended/popular list and also allows us to show `Sollet` which is web based and therefore not detectable.

- Screenshot of `Connect` button with wallet list:

Screen Shot 2022-03-17 at 2 53 53 PM

### CODE REVIEW

File: pages/account.tsx [hunk 1]
```diff
@@ -153,12 +150,6 @@ export default function Account() {
     }
   }, [pubkey, mangoGroup])
 
-  useEffect(() => {
-    if (connected) {
-      router.push('/account')
-    }
-  }, [connected])
```
Comment (kierangillen) [inline pages/account.tsx:160]: I'm not sure what the purpose of this code block was for? Refreshing the page when the wallet connected? But this was creating an infinite loop using the new `connected` hook and seemed to be to be unnecessary.

File: stores/useMangoStore.tsx [hunk 2]
```diff
@@ -56,7 +56,7 @@ export const ENDPOINTS: EndpointInfo[] = [
 ]
 
 type ClusterType = 'mainnet' | 'devnet'
-const DEFAULT_MANGO_GROUP_NAME = process.env.NEXT_PUBLIC_GROUP || 'mainnet.1'
+const DEFAULT_MANGO_GROUP_NAME = process.env.NEXT_PUBLIC_GROUP || 'devnet.2'
```
Comment (mschneider) [inline stores/useMangoStore.tsx:59]: shouldn't be updated

**Comment by kierangillen** [comment]
@mschneider Thanks yeah, the build is broken because I need to update the `WalletAdapter` type in the Client. PR for that is here: https://github.com/blockworks-foundation/mango-client-v3/pull/72 - sort of a chicken or the egg problem there.

**Comment by kierangillen** [comment]
Moving this PR to draft as we work out the UX/UI

File: @types/types.tsx [hunk 3]
```diff
@@ -124,13 +124,11 @@ export const DEFAULT_PUBLIC_KEY = new PublicKey(
 
 export interface WalletAdapter {
```
Comment (tjshipe) [inline @types/types.tsx:125]: Can we get rid of this type and just import the WalletAdapter type from the solana pkg?

File: components/WalletProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+          publicKey,
+          connected: true,
+          signTransaction,
+          signAllTransactions,
+          connect,
+          disconnect,
+        }
+      })
+
+      await actions.fetchMangoGroup()
```
Comment (tjshipe) [inline components/WalletProvider.tsx:51]: This shouldn't be needed here. The mango group should be loaded on page load.

File: components/index.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
+export * from './AccountInfo'
```
Comment (tjshipe) [inline components/index.tsx:1]: This is great, thanks for adding 🙏

File: pages/account.tsx (hunk 1 above)
Comment (tjshipe) [reply pages/account.tsx:160]: I think this may be needed for when you're viewing someone elses mango account. Lets sync up on this one

**Review by tjshipe — CHANGES_REQUESTED** [review]

File: @types/types.tsx (hunk 3 above)
Comment (kierangillen) [reply @types/types.tsx:125]: Unfortunately not yet. For now I'm coercing the WalletAdapter to match this type, but they don't match. Once we transition completely do using the WalletAdapter we can.

File: components/WalletSuggestionModal.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { useWallet } from '@solana/wallet-adapter-react'
+import React from 'react'
+import Modal from './Modal'
+
+interface Props {
+  onClose: () => void
+  isOpen: boolean
+}
+
+export const WalletSuggestionModal: React.FC<Props> = ({ isOpen, onClose }) => {
```
Comment (kierangillen) [inline components/WalletSuggestionModal.tsx:10]: @saml33 Here's the `WalletSuggestionModal` we discussed.

Comment (saml33) [reply components/WalletSuggestionModal.tsx:10]: Cool, thanks.

An issue with only showing detected wallets is it won't show web-based (eg Sollet web). Maybe in the dropdown we need a small list of detected/popular and the last item is "More Options" which when clicked closes the dropdown and opens a modal with a complete list. Could also have an expandable section in the modal "Don't have a wallet?" where we could make recommendations. What do you think?

Also, before interacting with the button it should probably have something selected by default. I clicked the Connect button and nothing happens which is maybe more confusing that it trying to connect to a wallet I don't intend to use. It says the wallet name under connect anyway so likely users will see that and change it before connecting

Comment (kierangillen) [reply components/WalletSuggestionModal.tsx:10]: I think for now let's include the wallets that aren't extensions, that are common, always and then show first the wallets that are installed, that covers all of our bases. We can later include the modal but I don't want to add more scope to this if not needed now.

**Pull request merged**